web/src/lib/components/pages.tsx

```diff
@@ -237,47 +237,23 @@ export default class Pages extends Component<PagesProps, PagesState> {
     });
   }
 
-  private uploadRecordings(recordings: any[],
-                           sentences: string[],
-                           progressCb: Function): Promise<void> {
-
-    return new Promise<void>((res, rej) => {
-      this.ensurePrivacyAgreement().then(() => {
-        let runningTotal = 0;
-        let originalTotal = recordings.length;
-
-        // This function calls itself recursively until
-        // all recordings are uploaded.
-        let uploadNext = () => {
-          if (recordings.length === 0) {
-            this.props.api.uploadDemographicInfo().then(() => {
-              return;
-            });
-            res();
-            return;
-          }
-
-          let recording = recordings.pop();
-          let blob = recording.blob;
-          let sentence = sentences.pop();
-
-          this.props.api.uploadAudio(blob, sentence).then(() => {
-            runningTotal++;
-            let percentage = Math.floor((runningTotal / originalTotal) * 100);
-            progressCb && progressCb(percentage);
-            this.props.user.tallyRecording();
-            uploadNext();
-          });
-        };
-
-        // Start the recursive chain to upload the recordings serially.
-        uploadNext();
-      }).catch(rej);
-    }).then(() => {
-      this.setState({
-        robot: ''
-      });
-    });
+  private async uploadRecordings(recordings: any[],
+                                 sentences: string[],
+                                 progressCb: Function): Promise<void> {
+    await this.ensurePrivacyAgreement();
+    const originalTotal = recordings.length;
+
+    for (let runningTotal = 1; runningTotal <= originalTotal; runningTotal++) {
+      const recording = recordings.pop();
+      const blob = recording.blob;
+      const sentence = sentences.pop();
+
+      await this.props.api.uploadAudio(blob, sentence);
+      let percentage = Math.floor((runningTotal / originalTotal) * 100);
```

Comment:
To keep the same functionality as before, we need to not do the progress/tally work when we have uploaded all the clips. For simplicity, let's add a check here for `recordings.length === 0` and if so let's not execute these three lines.